DotNetCore/MVC/CostConstruct/ConstructionApplication/Controllers/DailyAttendanceController.cs

```diff
@@ -91,12 +89,30 @@ public IActionResult Add(DailyAttendanceVm dailyAttendanceVm)
             List<JobCategory> jobCategories = _jobCategoryRepository.GetAll();
             ViewBag.JobCategory = new SelectList(jobCategories, "Id", "Name");
 
-            if (dailyAttendanceVm.JobCategoryId == 0 || dailyAttendanceVm.TotalWorker == 0 || dailyAttendanceVm.TotalWorker <= 0)
+            if (dailyAttendanceVm.JobCategoryId == 0 ||
+                dailyAttendanceVm.TotalWorker == 0 || 
+                dailyAttendanceVm.TotalWorker <= 0)
             {
                 ViewBag.errorMessage = "Please provide valid inputs. Job Category and Total Worker are required.";
                 return View(dailyAttendanceVm);
             }
 
+            if (dailyAttendanceVm.Date > DateTime.Now)
+            {
+                ViewBag.errorMessage = "Future dates are not allowed.";
+                return View(dailyAttendanceVm);
+            }
+
+            string totalWorkerPattern = @"^\d+$";
```

Comment:
This regex accept only numeric value, no special character and no alphabets are allow.